src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonFormatWriterGenerator.cs

```diff
@@ -187,7 +187,15 @@ private void InvokeOnSerialized(ClassDataContract classContract)
             private void WriteClass(ClassDataContract classContract)
             {
                 InvokeOnSerializing(classContract);
-                WriteMembers(classContract, null, classContract);
+                if (classContract.IsISerializable)
+                {
+                    _ilg.Call(_contextArg, JsonFormatGeneratorStatics.WriteJsonISerializableMethod, _xmlWriterArg, _objectLocal);
+                }
+                else
+                {
```

Comment:
There's some code missing in the else clause here to do with handling IExtensibleDataObject. This is currently in the .Net Standard 2.0 api, but not in 1.7 so I don't know if this needs to be pulled in too.